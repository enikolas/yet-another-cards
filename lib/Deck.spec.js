'use strict'

const Deck = require('./Deck')

describe('Deck', () => {
  it('should create a deck', () => {
    const deck = new Deck()

    expect(deck).toMatchSnapshot()
  })

  it('should shuffle the deck', () => {
    const deck = new Deck()
    deck.shuffle()

    expect(deck).not.toEqual(new Deck())
  })

  it('should draw a card from the deck', () => {
    const deck = new Deck()

    expect(deck.draw()).toMatchSnapshot()
    expect(deck).toMatchSnapshot()
  })

  it('should return readable string of a deck', () => {
    const deck = new Deck()

    expect(deck.toString()).toMatchSnapshot()
  })
})
