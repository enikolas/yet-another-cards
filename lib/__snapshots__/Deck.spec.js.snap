// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck should create a deck 1`] = `
Deck {
  "cards": Array [
    Card {
      "rank": "A",
      "suit": "♥",
    },
    Card {
      "rank": "K",
      "suit": "♥",
    },
    Card {
      "rank": "Q",
      "suit": "♥",
    },
    Card {
      "rank": "J",
      "suit": "♥",
    },
    Card {
      "rank": "T",
      "suit": "♥",
    },
    Card {
      "rank": "9",
      "suit": "♥",
    },
    Card {
      "rank": "8",
      "suit": "♥",
    },
    Card {
      "rank": "7",
      "suit": "♥",
    },
    Card {
      "rank": "6",
      "suit": "♥",
    },
    Card {
      "rank": "5",
      "suit": "♥",
    },
    Card {
      "rank": "4",
      "suit": "♥",
    },
    Card {
      "rank": "3",
      "suit": "♥",
    },
    Card {
      "rank": "2",
      "suit": "♥",
    },
    Card {
      "rank": "A",
      "suit": "♠",
    },
    Card {
      "rank": "K",
      "suit": "♠",
    },
    Card {
      "rank": "Q",
      "suit": "♠",
    },
    Card {
      "rank": "J",
      "suit": "♠",
    },
    Card {
      "rank": "T",
      "suit": "♠",
    },
    Card {
      "rank": "9",
      "suit": "♠",
    },
    Card {
      "rank": "8",
      "suit": "♠",
    },
    Card {
      "rank": "7",
      "suit": "♠",
    },
    Card {
      "rank": "6",
      "suit": "♠",
    },
    Card {
      "rank": "5",
      "suit": "♠",
    },
    Card {
      "rank": "4",
      "suit": "♠",
    },
    Card {
      "rank": "3",
      "suit": "♠",
    },
    Card {
      "rank": "2",
      "suit": "♠",
    },
    Card {
      "rank": "A",
      "suit": "♣",
    },
    Card {
      "rank": "K",
      "suit": "♣",
    },
    Card {
      "rank": "Q",
      "suit": "♣",
    },
    Card {
      "rank": "J",
      "suit": "♣",
    },
    Card {
      "rank": "T",
      "suit": "♣",
    },
    Card {
      "rank": "9",
      "suit": "♣",
    },
    Card {
      "rank": "8",
      "suit": "♣",
    },
    Card {
      "rank": "7",
      "suit": "♣",
    },
    Card {
      "rank": "6",
      "suit": "♣",
    },
    Card {
      "rank": "5",
      "suit": "♣",
    },
    Card {
      "rank": "4",
      "suit": "♣",
    },
    Card {
      "rank": "3",
      "suit": "♣",
    },
    Card {
      "rank": "2",
      "suit": "♣",
    },
    Card {
      "rank": "A",
      "suit": "♦",
    },
    Card {
      "rank": "K",
      "suit": "♦",
    },
    Card {
      "rank": "Q",
      "suit": "♦",
    },
    Card {
      "rank": "J",
      "suit": "♦",
    },
    Card {
      "rank": "T",
      "suit": "♦",
    },
    Card {
      "rank": "9",
      "suit": "♦",
    },
    Card {
      "rank": "8",
      "suit": "♦",
    },
    Card {
      "rank": "7",
      "suit": "♦",
    },
    Card {
      "rank": "6",
      "suit": "♦",
    },
    Card {
      "rank": "5",
      "suit": "♦",
    },
    Card {
      "rank": "4",
      "suit": "♦",
    },
    Card {
      "rank": "3",
      "suit": "♦",
    },
    Card {
      "rank": "2",
      "suit": "♦",
    },
  ],
}
`;

exports[`Deck should draw a card from the deck 1`] = `
Card {
  "rank": "2",
  "suit": "♦",
}
`;

exports[`Deck should draw a card from the deck 2`] = `
Deck {
  "cards": Array [
    Card {
      "rank": "A",
      "suit": "♥",
    },
    Card {
      "rank": "K",
      "suit": "♥",
    },
    Card {
      "rank": "Q",
      "suit": "♥",
    },
    Card {
      "rank": "J",
      "suit": "♥",
    },
    Card {
      "rank": "T",
      "suit": "♥",
    },
    Card {
      "rank": "9",
      "suit": "♥",
    },
    Card {
      "rank": "8",
      "suit": "♥",
    },
    Card {
      "rank": "7",
      "suit": "♥",
    },
    Card {
      "rank": "6",
      "suit": "♥",
    },
    Card {
      "rank": "5",
      "suit": "♥",
    },
    Card {
      "rank": "4",
      "suit": "♥",
    },
    Card {
      "rank": "3",
      "suit": "♥",
    },
    Card {
      "rank": "2",
      "suit": "♥",
    },
    Card {
      "rank": "A",
      "suit": "♠",
    },
    Card {
      "rank": "K",
      "suit": "♠",
    },
    Card {
      "rank": "Q",
      "suit": "♠",
    },
    Card {
      "rank": "J",
      "suit": "♠",
    },
    Card {
      "rank": "T",
      "suit": "♠",
    },
    Card {
      "rank": "9",
      "suit": "♠",
    },
    Card {
      "rank": "8",
      "suit": "♠",
    },
    Card {
      "rank": "7",
      "suit": "♠",
    },
    Card {
      "rank": "6",
      "suit": "♠",
    },
    Card {
      "rank": "5",
      "suit": "♠",
    },
    Card {
      "rank": "4",
      "suit": "♠",
    },
    Card {
      "rank": "3",
      "suit": "♠",
    },
    Card {
      "rank": "2",
      "suit": "♠",
    },
    Card {
      "rank": "A",
      "suit": "♣",
    },
    Card {
      "rank": "K",
      "suit": "♣",
    },
    Card {
      "rank": "Q",
      "suit": "♣",
    },
    Card {
      "rank": "J",
      "suit": "♣",
    },
    Card {
      "rank": "T",
      "suit": "♣",
    },
    Card {
      "rank": "9",
      "suit": "♣",
    },
    Card {
      "rank": "8",
      "suit": "♣",
    },
    Card {
      "rank": "7",
      "suit": "♣",
    },
    Card {
      "rank": "6",
      "suit": "♣",
    },
    Card {
      "rank": "5",
      "suit": "♣",
    },
    Card {
      "rank": "4",
      "suit": "♣",
    },
    Card {
      "rank": "3",
      "suit": "♣",
    },
    Card {
      "rank": "2",
      "suit": "♣",
    },
    Card {
      "rank": "A",
      "suit": "♦",
    },
    Card {
      "rank": "K",
      "suit": "♦",
    },
    Card {
      "rank": "Q",
      "suit": "♦",
    },
    Card {
      "rank": "J",
      "suit": "♦",
    },
    Card {
      "rank": "T",
      "suit": "♦",
    },
    Card {
      "rank": "9",
      "suit": "♦",
    },
    Card {
      "rank": "8",
      "suit": "♦",
    },
    Card {
      "rank": "7",
      "suit": "♦",
    },
    Card {
      "rank": "6",
      "suit": "♦",
    },
    Card {
      "rank": "5",
      "suit": "♦",
    },
    Card {
      "rank": "4",
      "suit": "♦",
    },
    Card {
      "rank": "3",
      "suit": "♦",
    },
  ],
}
`;

exports[`Deck should return readable string of a deck 1`] = `
Array [
  "A♥",
  "K♥",
  "Q♥",
  "J♥",
  "T♥",
  "9♥",
  "8♥",
  "7♥",
  "6♥",
  "5♥",
  "4♥",
  "3♥",
  "2♥",
  "A♠",
  "K♠",
  "Q♠",
  "J♠",
  "T♠",
  "9♠",
  "8♠",
  "7♠",
  "6♠",
  "5♠",
  "4♠",
  "3♠",
  "2♠",
  "A♣",
  "K♣",
  "Q♣",
  "J♣",
  "T♣",
  "9♣",
  "8♣",
  "7♣",
  "6♣",
  "5♣",
  "4♣",
  "3♣",
  "2♣",
  "A♦",
  "K♦",
  "Q♦",
  "J♦",
  "T♦",
  "9♦",
  "8♦",
  "7♦",
  "6♦",
  "5♦",
  "4♦",
  "3♦",
  "2♦",
]
`;
